### ISSUE TITLE
Combat Module: Weapons sort of changing to the levels of weapons beside them when opening the inventory and picking the weapon up out of the hotbar at the same time.

### ISSUE THREAD

**Issue by UltimasPrime** [daleao/modular-overhaul#152]
This is a weird bug/incompatibility that I noticed only because I play with combat module of your mod and Stardew Druid since that mod kind of forces you to play with multiple weapons at the same time in combat. I am not sure what is causing it and saw no traces of it in the log but if you scroll from one weapon in your hotbar to another and pick up the new weapon to move it at the same time while opening the inventory then it will sort of copy the weapons levels and stats to an extent. Due to this bug I turned my lava katana into a level 4 sword with lowered damage to start with since I had templars blade to the left of it. Then I changed it to a level 3 blade with even lower damage due to having a tempered broadsword to left of it. I sort of fixed it by placing the level 8 ruined blade from your mod to the left of it and preforming the same bug, but I noticed the level did not seem to go up but the damage did. I think what is happening is that the stardew druid mod has abilities that count as sword attacks and when you switch between them super fast and pick up the sword mid attack it somehow tricks the combat module from your mod into thinking your picking up a new version of the weapon based on the stats of the weapon that was just scrolled over. It is really hard to do intentionally, but during the chaos of combat it can be really easy to do sometimes. The last log that occurred right as it happened is [10:55:07 TRACE SpaceCore] Event: AddWalletItems. Here is the log for me changing the sword to a ruined blades stats, but I don't think it shows much:
[SMAPI-latest.txt](https://github.com/daleao/modular-overhaul/files/13928689/SMAPI-latest.txt)

**Comment by UltimasPrime** [comment]
Also the smapi log shows that I kept using the analyze spell from the magic mod, but you can disregard that. I have the button for that spell right beside one of the buttons that used to scroll my hotbar and kept hitting it by accident while trying to recreate the gllitch. The glitch first happened way before I even unlocked magic.

**Comment by daleao** [comment]
Sorry, the steps are quite confusing. Would you please list them out in order to make it more clear?
Ex.
1. Have Weapon A and Weapon B in adjacent inventory slots, and be holding weapon A.
2. Summon monster.
3. Press attack then quickly scroll to weapon B?
4. Move something to where??
5. Open inventory???
6. etc.

**Comment by UltimasPrime** [comment]
Sorry about that. This is probably still going to sound confusing because there are a lot of steps happening at the same time. It is probably a lot easier for me to reproduce it since I am using a controller with controller companion support which lets me input keys a lot faster than on keyboard. The steps to recreate the glitch are:

1. Have 2 weapons side by side in the tool bar (in my case it was the lava katana and your version of the dark sword).
2. Switch between the weapon whos level/stats you want to imprint to the weapon you want to be imprint while at the same time selecting the weapon you want to imprint and opening the inventory while your cursor is hovering the place in the inventory where weapon that you want imprinted will be and left click that spot which will automatically pick it up. (I did this by accident a lot since I was opening inventory to pause during battle to rearrange my hot bar because some of the stardew druid boss fights/encounters are extremely difficult)
3. The weapon's stats are imprinted on the other weapon.

**Comment by daleao** [comment]
Yeah I need you to break that middle step down in bite-sized chunks.

- Select Weapon A (Slot 1)
- Scroll to Weapon B (Slot 2)
- Press hotkey to switch back to Slot 1
- Open inventory
- Right click (mouse must be coincidentally hovering over Weapon B?)

Is that correct?
Might be easier if you could record a video.

**Comment by UltimasPrime** [comment]
Yeah that is right. It is really difficult to do on purpose.

**Comment by UltimasPrime** [comment]
It should cause you to have weapon picked up while in your inventory, and when you place it back down it has the updated levels/stats.

**Comment by daleao** [comment]
Since it's so difficult and unlikely to reproduce, it'll probably be easier to offer a workaround instead.
Can you try selecting the glitched weapon and entering `cmbt refresh` in the console?

**Comment by UltimasPrime** [comment]
Well I tried that but the problem is that because it is now a level 3 lava katana it only changes its stats to match a level 3 weapon.

**Comment by UltimasPrime** [comment]
I can probably do the glitch again and use the console command again to fix it though.

**Comment by UltimasPrime** [comment]
Well I tried that and for some reason it undid the glitch that sort of fixed it and turned the lava katana back to a level 3 sword.

**Comment by UltimasPrime** [comment]
If you can't reproduce it might just be because of a mod incompatibility with stardew druid.
